pkg/pointerdb/pointerdb_test.go

```diff
@@ -143,92 +147,176 @@ func TestServiceDelete(t *testing.T) {
-		{"", "", "videos", false, 0, meta.None, nil, keys, keys[:2], false, nil, ""},
-		{"music", "", "", false, 0, meta.None, nil, keys[2:], storage.Keys{keys[2], keys[3], keys[5]}, false, nil, ""},
-		{"music", "", "", true, 0, meta.None, nil, keys[2:], keys[2:6], false, nil, ""},
-		{"music", "song1.mp3", "", true, 0, meta.None, nil, keys, keys[3:6], false, nil, ""},
-		{"music", "song1.mp3", "album/song3.mp3", true, 0, meta.None, nil, keys, keys[3:4], false, nil, ""},
-		{"music", "", "song4.mp3", true, 0, meta.None, nil, keys, keys[2:5], false, nil, ""},
-		{"music", "", "song4.mp3", true, 1, meta.None, nil, keys, keys[4:5], true, nil, ""},
-		{"music", "", "song4.mp3", false, 0, meta.None, nil, keys, keys[2:4], false, nil, ""},
-		{"music", "song2.mp3", "song4.mp3", true, 0, meta.None, nil, keys, keys[4:5], false, nil, ""},
-		{"mus", "", "", true, 0, meta.None, nil, keys[1:], nil, false, nil, ""},
-	} {
-		errTag := fmt.Sprintf("Test case #%d", i)
+	pointer := &pb.Pointer{}
+	pointer.CreationDate = ptypes.TimestampNow()
 
-		db := NewMockKeyValueStore(ctrl)
-		s := Server{DB: db, logger: zap.NewNop()}
+	pointerBytes, err := proto.Marshal(pointer)
+	if err != nil {
+		t.Fatal(err)
+	}
+	pointerValue := storage.Value(pointerBytes)
+
+	err = storage.PutAll(db, []storage.ListItem{
+		{Key: key("sample.😶"), Value: pointerValue},
+		{Key: key("müsic/söng1.mp3"), Value: pointerValue},
+		{Key: key("müsic/söng2.mp3"), Value: pointerValue},
+		{Key: key("müsic/album/söng3.mp3"), Value: pointerValue},
+		{Key: key("müsic/söng4.mp3"), Value: pointerValue},
+		{Key: key("ビデオ/movie.mkv"), Value: pointerValue},
+	}...)
+	if err != nil {
+		t.Fatal(err)
+	}
 
-		if tt.err != nil || tt.errString == "" {
-			prefix := storage.Key([]byte(tt.prefix + "/"))
-			db.EXPECT().List(prefix, storage.Limit(0)).Return(tt.returnedKeys, tt.err)
-
-			if tt.metaFlags != meta.None {
-				pr := pb.Pointer{}
-				b, err := proto.Marshal(&pr)
-				assert.NoError(t, err, errTag)
-				for _, key := range keys {
-					db.EXPECT().Get(key).Return(b, nil)
-				}
+	type Test struct {
+		Request  pb.ListRequest
+		Expected *pb.ListResponse
+		Error    func(i int, err error)
+	}
+
+	errorWithCode := func(code codes.Code) func(i int, err error) {
+		t.Helper()
+		return func(i int, err error) {
+			t.Helper()
+			if status.Code(err) != code {
+				t.Fatalf("%d: should fail with %v, got: %v", i, code, err)
 			}
 		}
+	}
 
-		req := pb.ListRequest{
-			Prefix:     tt.prefix,
-			StartAfter: tt.startAfter,
-			EndBefore:  tt.endBefore,
-			Recursive:  tt.recursive,
-			Limit:      int32(tt.limit),
-			MetaFlags:  tt.metaFlags,
-			APIKey:     tt.apiKey,
-		}
-		resp, err := s.List(ctx, &req)
+	tests := []Test{
+		{
+			Request: pb.ListRequest{Recursive: true},
+			Expected: &pb.ListResponse{
+				Items: []*pb.ListResponse_Item{
+					{Path: "müsic/album/söng3.mp3"},
+					{Path: "müsic/söng1.mp3"},
+					{Path: "müsic/söng2.mp3"},
+					{Path: "müsic/söng4.mp3"},
```

Comment:
Let's also have an object with name "müsic" and make sure it is not picked by up when listing with prefix "müsic".